File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        result = []
        
        def backtrack(i, currsum, path):
            if (currsum > target):
                return
            if (target == currsum):
                result.append(path.copy())
                return
            if (i >= len(nums)):
                return
            
            #add current again
            path.append(nums[i])
            backtrack(i, currsum+nums[i], path)

            #add next one
            path.pop()
            backtrack(i+1, currsum, path)
            
        backtrack(0, 0, [])
        return result


            
            

File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    public int climbStairs(int n) {
        int one = 0;
        int two = 1;
        for (int i = 0; i < n; i++){
            int temp = one;
            one = two;
            two = one+temp;
        }
        return two;
    }
}
